src/ORM/ResultSet.php

```diff
@@ -313,10 +313,19 @@ public function first()
      */
     public function serialize()
     {
+        if (!$this->_useBuffering) {
+            $msg = 'You cannot serialize an un-buffered ResultSet. Use Query::bufferResults() to get a buffered ResultSet.';
+            throw new Exception($msg);
+        }
+
         while ($this->valid()) {
             $this->next();
         }
 
+        if ($this->_results instanceof SplFixedArray) {
```

Comment:
I'm considering just removing SplFixedArray... it did not perform as good as I wanted it to